include/threads/ThreadState.h

```diff
@@ -21,31 +21,34 @@ class Level
     uint32_t id;
 
     // Number of nested OpenMP constructs
-    int depth = 0;
+    int32_t depth = 0;
 
     // Number of parallel regions with more than 1 thread above this level
-    int activeLevels = 0;
+    int32_t activeLevels = 0;
 
     // Max number of active parallel regions allowed
-    int maxActiveLevels = 1;
+    int32_t maxActiveLevels = 1;
 
     // Number of threads of this level
-    int numThreads = 1;
+    int32_t numThreads = 1;
 
     // Desired number of thread set by omp_set_num_threads for all future levels
-    int wantedThreads = -1;
+    int32_t wantedThreads = -1;
 
     // Num threads pushed by compiler, valid for one parallel section.
     // Overrides wantedThreads
-    int pushedThreads = -1;
+    int32_t pushedThreads = -1;
 
     // Offset for the global thread numbers at this level
     int32_t globalTidOffset = 0;
 
     uint32_t nSharedVarOffsets = 0;
     uint32_t* sharedVarOffsets;
 
-    Level(int numThreadsIn);
+    static std::shared_ptr<Level> deserialise(
+      const std::vector<uint8_t>& bytes);
```

Comment:
Previously `deserialise` was an instance method on the `Level` class (i.e. create the `Level`, _then_ deserialise the contents from some bytes), but it makes more sense to have it as a static factory-like method.